docs/general/developer-resources/sapphire-paratime/writing-dapps-on-sapphire.md

```diff
@@ -0,0 +1,183 @@
+on the other hand, should be considered unstable software and may have its state wiped at any
+time. As the name implies, only use the Testnet for testing unless you're testing how angry
+your users get when state is wiped.
+
+:::danger Never deploy production services on Testnet
+
+Because Testnet state can be wiped in the future, you should **never** deploy a
+production service on the Testnet! Just don't do it!
+
+:::
+
+:::tip
+
+For testing purposes, visit our [Testnet faucet] to obtain some TEST which you
+can then use on the Sapphire Testnet to pay for gas fees. The faucet supports
+sending TEST both to your consensus layer address or to your address inside the
+ParaTime.
+
+:::
+
+[network-parameters]: ../../oasis-network/network-parameters.md
+[Testnet]: ../../foundation/testnet/README.md
+[Sapphire Mainnet]: ./README.mdx#mainnet
+[Sapphire Testnet]: ./README.mdx#testnet
+
+## Sapphire vs Ethereum
+
+The Sapphire ParaTime is generally compatible with Ethereum, the EVM, and all of the
+user and developer tooling that you already use. There are a few breaking changes,
+but we think that you'll like them:
+
+* Contract state is only visible to the contract that wrote it. With respect
+  to the contract API, it's as if all state variables are declared as `private`, but
+  with the further restriction that not even full nodes can read the values. Public or
+  access-controlled values are provided instead through explicit getters.
+* Transactions and calls are end-to-end encrypted into the contract. Only the caller
+  and the contract can see the data sent to/received from the ParaTime. This ends up
+  defeating most of the utility of block explorers, however.
+* The `from` address using of calls is derived from a signature attached to the call.
+  Unsigned calls have their sender set to the zero address. This allows contract authors
+  to write getters that release secrets to authenticated callers, but without without
+  requiring a transaction to be posted on-chain.
+
+With confidentiality, you get a few extra benefits including the ability to generate private
+entropy, and make signatures on-chain. An example of a DApp that uses both is a HSM contract
+that generates an Ethereum wallet and signs transactions sent to it via transactions.
+
+Otherwise Sapphire is like Emerald, which is like a fast, cheap Ethereum.
+
+## Integrating Sapphire
+
+Once ROSE tokens are [deposited into Sapphire], it should be painless for users to begin
+using DApps. To achieve this ideal user experience, we have to modify the DApp a little,
+but it's made simple by our compatibility library, [`@oasislabs/sapphire-paratime`].
+Please refer to the library docs for more information on its use.
+
+There are compatibility layers in other languages, which may be found in [the repo].
+
+
+[deposited into Sapphire]: ../../manage-tokens/how-to-transfer-rose-into-evm-paratime.mdx
+[`@oasislabs/sapphire-paratime`]: https://www.npmjs.com/@oasislabs/sapphire-paratime
```

Comment:
We should probably mark this as "coming soon" for now, so we can go ahead and publish these docs.